feat: Add oxidized Result support for fallible functions

rlch opened the pull request:
Hey @fzyzcjy 

This PR adds support for serializing `Result<T, E>` to the equivalent exposed by [oxidized](https://pub.dev/packages/oxidized). I think the package's API has stabilized, obviously idiomatic w.r.t. Rust conventions; it's well-tested and doesn't pull in a crap-ton of API's like a lot of the sibling packages.

Thanks again for this package, I hope to contribute more over the coming years; what you've done is magnificent and IMO the best addition to the Flutter ecosystem thus far. 

## Changes

Fixes #2683 

## Procedure and Checklist

In order to quickly iterate and avoid slowing down development pace by making CI pass, only the following simplified steps are needed, and I (fzyzcjy) will handle the rest of CI / moving the tests currently (will have more automation in the future).

- [x] Implement the feature / fix the bug.
- [x] Add tests in `frb_example/dart_minimal`.
- [x] Make `dart_minimal`'s CI green.

Utility commands
- Run codegen: `cargo run --manifest-path /path/to/flutter_rust_bridge/frb_codegen/Cargo.toml -- generate`
- Run tests: `./frb_internal test-dart-native --package frb_example/dart_minimal`

fzyzcjy committed:
feat: add oxidized Result support for fallible functions

When oxidized package is detected in pubspec.yaml dependencies:
- Fallible Rust functions return Future<Result<T, E>> instead of Future<T>
- Errors are returned as Err(...) instead of throwing exceptions
- Panics still throw PanicException (bugs, not expected errors)

Runtime (frb_dart):
- Add decodeAsResult() to SimpleDecoder for Result-based decoding
- Add decodeObjectAsResult/decodeWireSyncTypeAsResult to BaseCodec
- Add executeNormalAsResult/executeSyncAsResult to BaseHandler
- Export Result, Ok, Err from flutter_rust_bridge_for_generated_common.dart

Codegen (frb_codegen):
- Add has_dependency() to DartRepository
- Add detect_oxidized_dependency() in generator_parser.rs
- Add use_oxidized field to GeneratorApiDartInternalConfig
- Wrap fallible function return types in Result<T, E>
- Use executeNormalAsResult/executeSyncAsResult for fallible functions

Tested with dart_minimal example:
- Async functions: Future<Result<T, E>>
- Sync functions: Result<T, E>

rlch committed:
feat: resolve generic type aliases for Result error types

  When a type alias like \`type Result<T> = std::result::Result<T, MyError>\`
  is used, FRB now correctly extracts the error type from the alias definition
  instead of falling back to AnyhowException.

  - Store generic type aliases separately in HirFlatTypeAlias
  - Look up generic Result aliases when parsing function return types
  - Substitute type parameters to resolve the actual error type

rlch committed:
feat: add test examples for generic Result type alias resolution

- Add example error types and functions using type aliases
- Update generated Dart/Rust bindings with new test cases

fzyzcjy reviewed (commented):
tiny nits (I will review more ASAP when having time)

rlch commented on frb_codegen/src/library/codegen/config/internal_config_parser/generator_parser.rs:
```diff
@@ -56,6 +58,7 @@ pub(super) fn parse(args: Args) -> anyhow::Result<GeneratorInternalConfig> {
     let default_external_library_loader =
         compute_default_external_library_loader(rust_crate_dir, dart_root, config);
     let c_symbol_prefix = compute_c_symbol_prefix(dart_root)?;
+    let use_oxidized = detect_oxidized_dependency(dart_root);
```
Yeah I was tossing that up. Will do!

rlch committed:
feat(codegen): add #[frb(oxidized)] attribute for per-function oxidized Result support

rlch committed:
feat: demonstrate per-function #[frb(oxidized)] opt-in behavior

rlch committed:
Merge branch 'master' into result-oxidized

rlch committed:
fix: filter traits with #[frb(ignore)] in traits_map()

Traits marked with #[frb(ignore)] were being collected into src_traits
but the ignore attribute was never checked, causing panics when the
trait was used in Option<T> contexts.

This mirrors how structs/enums handle the ignore attribute via
parse_struct_or_enum_should_ignore.

Amp-Thread-ID: https://ampcode.com/threads/T-019bf8ee-f69c-75eb-981d-55743289c99e
Co-authored-by: Amp <amp@ampcode.com>

rlch committed:
fix: handle trait object parsing and ignored trait filtering in codegen

rlch committed:
feat: add translatable serde_json::Value support

Add first-class support for serde_json::Value as a delegate type that
serializes via JSON string across the FFI boundary, mapping to Dart's
Object?. Supports all codecs (SSE, DCO, CST) and composite types
(Option, Vec, HashMap, struct fields). Includes tests, documentation,
and feature flag gated behind `serde_json`.

fzyzcjy committed:
Move oxidized Result coverage to pure Dart

fzyzcjy committed:
Revert "feat: add translatable serde_json::Value support"

This reverts commit baadddbf2575623693c80d31c7aea56529de76e1.

fzyzcjy committed:
Add pure Dart oxidized Result coverage

fzyzcjy committed:
Update internal lockfile for oxidized dependency

fzyzcjy committed:
Polish oxidized Result docs

fzyzcjy committed:
Remove unrelated example diffs

fzyzcjy committed:
Avoid oxidized dependency in Dart runtime

fzyzcjy committed:
Import referenced wire Rust namespaces

fzyzcjy committed:
Merge origin/master into result oxidized branch

fzyzcjy committed:
Simplify oxidized Result wrapping
